pkg/trace/sampler/remote_config_handler.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
This si all completely untested.

What would you recommend?

I could unit test this by mocking the remote client, but maybe that's lots of code for few added value.

Is there a way to do some integration/end to end testing?

Reply:
You'll want to end up doing some integration/end to end testing in staging but I don't think there's any good way to do it in CI/CD.

Thankfully you shouldn't have to go through one of the heavy-handed mock frameworks to be able to spin up some code to test this out. `Start()` and `Stop()` can just be nops. Register can just store the function pointer, and then you can write code to push an update through.

```
func (c *testRemoteClient) Start() {}
func (c *testRemoteClient) Stop() {}
func (c *testRemoteClient) RegisterAPMUpdate(fn <whatever the definition is>) {
   c.f = fn
}
func (c *testRemoteClient) pushUpdate(update map[string]state.APMSamplingConfig) {
    c.f(update)
}
```

Then the unit tests are just defining your update, calling pushUpdate on your testRemoteclient, and inspecting the PrioritySampler.

Reply:
Thanks for the pointers, I'll do that!